PROTOCOL.md

```diff
@@ -314,6 +320,7 @@ extendedFileMetadata | Boolean | When `true` the fields `partitionValues`, `size
 partitionValues| Map[String, String] | A map from partition column to value for this file. See also [Partition Value Serialization](#Partition-Value-Serialization)
 size| Long | The size of this file in bytes
 tags | Map[String, String] | Map containing metadata about this file
+numRecords | Option[Long] | The number of records in this file.
```

Comment:
I assume this description is correct. All the files I have checked it seems to be set to `null`. @scottsand-db could confirm.

Reply:
this is correct

Reply:
Ok. `numRecords` should not be written out to the checkpoint file, nor the delta log `.json` file. It is marked as `JsonIgnore` in the RemoveFile class declaration, and the fact that it is still being written out to the checkpoint file is a bug. I'll file an issue for this :)

Reply:
Does that mean we should just close this PR then?

Reply:
Why don't you rename this PR to `Minor improvements to protocol.md table of contents`

Reply:
@scottsand-db if numRecords is not supposed to be written out to both the checkpoint and commit log, then what's the reason for keeping it in the spec? This sounds like an implementation details specific to in memory data structure to me?

Reply:
> then what's the reason for keeping it in the spec?

It should not be written to the spec.

Reply:
We should remove that field from this PR, and update this PR to `Minor improvements to protocol.md table of contents`. Thanks for asking/clarifying!

Reply:
Removed